src/main/scala/com/crealytics/spark/v2/excel/ExcelHelper.scala

```diff
@@ -132,11 +134,24 @@ class ExcelHelper private (options: ExcelOptions) {
     * @return
     *   cell-row iterator
     */
-  def getRows(conf: Configuration, uri: URI): Iterator[Vector[Cell]] = {
+  def getRows(conf: Configuration, uri: URI): CloseableIterator[Vector[Cell]] = {
     val workbook = getWorkbook(conf, uri)
     val excelReader = DataLocator(options)
-    try { excelReader.readFrom(workbook) }
-    finally workbook.close()
+    try {
+      val rowIter = excelReader.readFrom(workbook)
+      workbook match {
+        case _: StreamingWorkbook => CloseableIterator(rowIter, Seq(workbook))
+        case _ => {
+          workbook.close()
```

Comment:
why do you close the workbook here and do not hand that to CloseableIterator()? Would be more consistent to me

Reply:
this workbook doesn't need to be kept open but I suppose it could be changed to work like the streaming workbook